src/main/io/serial_cli.c

```diff
@@ -566,9 +566,9 @@ const clivalue_t valueTable[] = {
     { "align_acc",                  VAR_UINT8  | MASTER_VALUE | MODE_LOOKUP,  &masterConfig.sensorAlignmentConfig.acc_align, .config.lookup = { TABLE_ALIGNMENT } },
     { "align_mag",                  VAR_UINT8  | MASTER_VALUE | MODE_LOOKUP,  &masterConfig.sensorAlignmentConfig.mag_align, .config.lookup = { TABLE_ALIGNMENT } },
 
-    { "align_board_roll",           VAR_INT16  | MASTER_VALUE,  &masterConfig.boardAlignment.rollDegrees, .config.minmax = { -180,  360 } },
-    { "align_board_pitch",          VAR_INT16  | MASTER_VALUE,  &masterConfig.boardAlignment.pitchDegrees, .config.minmax = { -180,  360 } },
-    { "align_board_yaw",            VAR_INT16  | MASTER_VALUE,  &masterConfig.boardAlignment.yawDegrees, .config.minmax = { -180,  360 } },
+    { "align_board_roll",           VAR_INT16  | MASTER_VALUE,  &boardAlignment.rollDegrees, .config.minmax = { -180,  360 } },
+    { "align_board_pitch",          VAR_INT16  | MASTER_VALUE,  &boardAlignment.pitchDegrees, .config.minmax = { -180,  360 } },
```

Comment:
i was looking at cliPrintVar() and yes.  the addresses in valueTable you pointed out can be mis-aligned.

i verified this by placing a uint8_t between two of the uint16_t values in boardAlignment_t.

However, the code for cliGet and cliSet still works just fine for the mis-aligned 16 bit value which had a ptr value of 0x20004b0b <boardAlignment+3> in my test.

https://github.com/cleanflight/cleanflight/blob/configuration-storage/src/main/io/serial_cli.c#L2228

yes, the ptr adjustment based on current profile will certainly need work.

i need a little help here, can you explain further?

for reference I had a glance at this too:

https://www.securecoding.cert.org/confluence/display/c/EXP36-C.+Do+not+cast+pointers+into+more+strictly+aligned+pointer+types